### DISCUSSION TITLE
Access body in Middleware

### DISCUSSION THREAD

**Discussion by Alacho2** [tokio-rs/axum#1864] [Q&A] [answered]
- [x] I have looked for existing issues (including closed) about this

## Feature Request
Coming from Express in NodeJS, accessing a body in middleware is a trivial exercise of a one-liner. How do you do this in Axum? I was looking at the example but I couldn't find a nice way of doing this

**Comment by MartinMinkov** [comment, marked as answer]
Hey! This example shows the way of accessing the body in middleware: https://github.com/tokio-rs/axum/blob/main/examples/consume-body-in-extractor-or-middleware/src/main.rs

It depends on what you would like to do, but in this case, you define your `from_request` function to do whatever sort of middleware actions you want (e.g. return a custom extractor or validate.

I'm not sure if linking to youtube is encouraged on these issues but I found this tutorial video very helpful in setting up my middleware: https://www.youtube.com/watch?v=5KOr_ATiuAU
